### DISCUSSION TITLE
Question about Depth Estimation

### DISCUSSION THREAD

**Discussion by AlexanderJGomez** [luxonis/depthai#559] [Q&A]
Hello, I am currently using depth estimation with object detection to calculate the lat, longs of objects in a space and was hoping to get a sanity check on the logic of this process.

With the following device settings:
```python
        self.nodes.stereo.setRuntimeModeSwitch(True)
        self.nodes.stereo.setLeftRightCheck(True)
        self.nodes.stereo.setSubpixel(True)

        self.nodes.stereo.setDepthAlign(dai.CameraBoardSocket.RGB)
```

I take the bounding box from the color camera and scale it to the depth frame. Then, I use the scaled bounding box to calculate the average depth over that region in the depth frame. Using that average depth, I approximate x, y, z coordinates, similar to how it is done [here](https://github.com/luxonis/depthai-experiments/tree/4c59a2aa5f2849f2dd410e10981fdefb5fdda6dc/gen2-human-machine-safety).

Then, I calculate the absolute distance from the center of our camera to the point and the angle (before converting to lat long). I calculate the distance by doing the following:

```python
    dist = math.sqrt(x**2 + z**2) # ignores height of point
```

Lastly, given the angle to the object and the distance away it is (along with other information about where I am and where I am facing), I calculate the object's location.

**Comment by Erol444** [comment]
Hello @AlexanderJGomez , I believe the `Z` is already the distance to the object, and `X`and `Y` are then calculated from the `Z`.

**Comment by AlexanderJGomez** [comment]
Hi @Erol444  thanks for the response. I'm a bit confused because [this example](https://github.com/luxonis/depthai-experiments/blob/4c59a2aa5f2849f2dd410e10981fdefb5fdda6dc/gen2-human-machine-safety/main.py#L86) seems to indicate that z is just another dimension in a 3D cartesian space

**Reply by SzabolcsGergely** [reply]
Z is indeed just another dimension in a 3D cartesian space, the distance to the object is the Euclidean distance. `sqrt(x^2+y^2+z^2)`
Depth map basically returns the `Z` dimension, if you point the camera to a flat surface, all the values will be the same (given that surface has enough texture).
